lambdas/services/pds_api_service.py

```diff
@@ -31,32 +51,100 @@ def handle_response(self, response: Response, nhs_number: str) -> PatientDetails
 
         raise PdsErrorException("Error when requesting patient from PDS")
 
-    def fake_pds_request(self, nhsNumber: str) -> Response:
-        mock_pds_results: list[dict] = []
-
-        try:
-            with open("services/mock_data/pds_patient.json") as f:
-                mock_pds_results.append(json.load(f))
-
-            with open("services/mock_data/pds_patient_restricted.json") as f:
-                mock_pds_results.append(json.load(f))
-
-        except FileNotFoundError:
-            raise PdsErrorException("Error when requesting patient from PDS")
+    def pds_request(self, nshNumber: str, retry_on_expired: bool):
+        endpoint, access_token_response = self.get_parameters_for_pds_api_request()
```

Comment:
Minor gripe, feel free to ignore but what would you say to creating a pydantic model to represent the access_token_response? This way we can validate the response matches the model and don't need to stress about whether the dict keys we access below are present and try/catching `KeyError` exceptions